### PR TITLE
Price fetcher ENTSOE/RTE

### PR DESCRIPTION
**Pull request by nikkozzblu**

### CODE REVIEW

**Comment by corradio** [comment]
Thanks man!
Refers to #205 , will review soon.

**Comment by corradio** [comment]
Seems like you are fetching prices in the future (for FR at least):

```
feeder❯ python -c 'from parsers.FR import *; print fetch_prices("FR")'
{'source': 'rte-france.com', 'price': 94.98, 'countryCode': 'FR', 'datetime': datetime.datetime(2017, 1, 20, 23, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris'))}

feeder❯ python -c 'from datetime import datetime; print datetime.now()'
2017-01-20 21:47:07.401887
```

Best for now is to fetch the last price <= now.

File: feeder/feeder.py [hunk 1]
```diff
@@ -343,6 +380,20 @@ def fetch_exchanges():
             statsd.increment('fetch_one_exchange_error')
             logger.exception('Exception while fetching exchange of %s' % k)
 
+def fetch_prices():
```
Comment (corradio) [inline feeder/feeder.py]: I'd recommend naming this function (and other related functions) `fetch_price`. There's no reason to have it in plural.

File: feeder/feeder.py [hunk 2]
```diff
@@ -224,6 +224,41 @@
...
+    'NO': ENTSOE.fetch_prices,
+    'PL': ENTSOE.fetch_prices,
+    'PT': ENTSOE.fetch_prices,
+    'RO': ENTSOE.fetch_prices,
+    'RS': ENTSOE.fetch_prices,
+    'SE': ENTSOE.fetch_prices,
+    'SI': ENTSOE.fetch_prices,
+    'SK': ENTSOE.fetch_prices,
+}
+
```
Comment (corradio) [inline feeder/feeder.py]: Style: extra lines

File: feeder/feeder.py [hunk 3]
```diff
@@ -224,6 +224,41 @@
     # 'SK->UA':     ENTSOE.fetch_exchange,
 }
 
+PRICE_PARSERS = {
+    
+    'FR': FR.fetch_prices,
```
Comment (corradio) [inline feeder/feeder.py]: Can you order alphabetically?

File: feeder/parsers/FR.py [hunk 4]
```diff
@@ -62,5 +62,37 @@ def fetch_production(country_code='FR', session=None):
 
     return data
 
+def fetch_prices(country_code,session=None,from_date=None,to_date=None):
```
Comment (corradio) [inline feeder/parsers/FR.py]: Style: add spaces between arguments

File: feeder/parsers/FR.py [hunk 5]
```diff
@@ -62,5 +62,37 @@ def fetch_production(country_code='FR', session=None):
...
+    r = session or requests.session()
+    formatted_from = from_date or arrow.now(tz='Europe/Paris').format('DD/MM/YYYY')
```
Comment (corradio) [inline feeder/parsers/FR.py]: 👍

File: feeder/feeder.py [hunk 6]
```diff
@@ -224,6 +224,41 @@
     # 'SK->UA':     ENTSOE.fetch_exchange,
 }
 
+PRICE_PARSERS = {
+
```
Comment (corradio) [inline feeder/feeder.py]: Style: extra line

File: feeder/feeder.py [hunk 7]
```diff
@@ -343,6 +380,20 @@ def fetch_exchanges():
             statsd.increment('fetch_one_exchange_error')
             logger.exception('Exception while fetching exchange of %s' % k)
 
+def fetch_prices():
+    for country_code, parser in PRICE_PARSERS.iteritems():
+        try:
+            with statsd.StatsdTimer('fetch_one_price'):
+                obj = parser(country_code,session)
+                if not obj: continue
+                # validate_price(obj)
```
Comment (corradio) [inline feeder/feeder.py]: I'd remove that line

File: feeder/feeder.py [hunk 8]
```diff
@@ -343,6 +380,20 @@ def fetch_exchanges():
             statsd.increment('fetch_one_exchange_error')
             logger.exception('Exception while fetching exchange of %s' % k)
 
+def fetch_prices():
+    for country_code, parser in PRICE_PARSERS.iteritems():
+        try:
+            with statsd.StatsdTimer('fetch_one_price'):
+                obj = parser(country_code,session)
```
Comment (corradio) [inline feeder/feeder.py]: Style: add space between arguments

File: feeder/parsers/FR.py [hunk 9]
```diff
@@ -62,5 +62,37 @@ def fetch_production(country_code='FR', session=None):
...
+
+    prices = []
+    datetimes = []
+
+    date_str = mixtr.get('date')
+    date = arrow.get(arrow.get(date_str).datetime, 'Europe/Paris')
+    for country_item  in mixtr.getchildren():
+        if country_item.get('granularite') != 'Global': continue
+        country_c=country_item.get('perimetre')
+        if(country_code!=country_c): continue
```
Comment (corradio) [inline feeder/parsers/FR.py]: Style: remove parenthesis, add space around operator `!=`

File: feeder/parsers/FR.py [hunk 10]
```diff
@@ -62,5 +62,37 @@ def fetch_production(country_code='FR', session=None):
...
+    response = r.get(url)
+    obj = ET.fromstring(response.content)
+    mixtr = obj[5]
+
+    prices = []
+    datetimes = []
+
+    date_str = mixtr.get('date')
+    date = arrow.get(arrow.get(date_str).datetime, 'Europe/Paris')
+    for country_item  in mixtr.getchildren():
```
Comment (corradio) [inline feeder/parsers/FR.py]: Style: extra space before `in`

File: feeder/parsers/FR.py [hunk 11]
```diff
@@ -62,5 +62,37 @@ def fetch_production(country_code='FR', session=None):
...
+        if country_item.get('granularite') != 'Global': continue
+        country_c=country_item.get('perimetre')
+        if(country_code!=country_c): continue
+        value = None
+        for value in country_item.getchildren():
+            if(value.text=='ND'): continue
```
Comment (corradio) [inline feeder/parsers/FR.py]: Style: Remove parenthesis and add space around `==`

**Comment by nikkozzblu** [comment]
Ok this fixed, I simply ignore the timestamp in future and review default period for FR to last 24h rather than next 24h,

File: feeder/parsers/FR.py [hunk 12]
```diff
@@ -62,5 +62,41 @@ def fetch_production(country_code='FR', session=None):
 
     return data
 
+def fetch_price(country_code, session=None, from_date=None, to_date=None):
+    r = session or requests.session()
+    dt_now=arrow.now(tz='Europe/Paris')
+    formatted_from = from_date or dt_now.replace(days=-1).format('DD/MM/YYYY')
```
Comment (corradio) [inline feeder/parsers/FR.py]: If you set formatted_from = formatted_to = `dt_now.format('DD/MM/YYYY')` then you actually get the last data point (this is the same behavior as in fetch production. Else it doesn't work (I get the last datapoint of yesterday instead).

Weird API behavior though.

File: feeder/parsers/FR.py [hunk 13]
```diff
@@ -62,5 +62,41 @@ def fetch_production(country_code='FR', session=None):
 
     return data
 
+def fetch_price(country_code, session=None, from_date=None, to_date=None):
+    r = session or requests.session()
+    dt_now=arrow.now(tz='Europe/Paris')
```
Comment (corradio) [inline feeder/parsers/FR.py]: Style: requires space around `=`

File: feeder/parsers/FR.py (hunk 12 above)
Comment (nikkozzblu) [inline feeder/parsers/FR.py]: Ok I implement the "last point only" as default behavior for FR price.
The issue with 'position' was fixed and comitted but might be I did it after merging

**Comment by nikkozzblu** [comment]
My mistake, fixed it in the wrong local copy, should be fine now

File: feeder/feeder.py [hunk 14]
```diff
@@ -224,6 +224,38 @@
...
+    'AT': ENTSOE.fetch_price,
+    'BE': ENTSOE.fetch_price,
+    'BG': ENTSOE.fetch_price,
+    'CH': ENTSOE.fetch_price,
+    'CZ': ENTSOE.fetch_price,
+    'DE': ENTSOE.fetch_price,
+    'DK': ENTSOE.fetch_price,
+    'EE': ENTSOE.fetch_price,
+    'ES': ENTSOE.fetch_price,
+    'FR': FR.fetch_price,
```
Comment (corradio) [inline feeder/feeder.py]: FR comes after FI :)

**Comment by nikkozzblu** [comment]
Hey,

Yes I can work on front-end, 
On my end I was thinking of some white circle centered on country containing prices (maybe with some border colors to reflect the high/low level relative to the rest)
But let know me know what you idea is :)

**Comment by corradio** [comment]
I think it will be problematic for small countries.
Let's discuss on #205

**Pull request merged**